### PR TITLE
layers: Fix HasExternalMemoryImportSupport with DRM modifiers

### PR DESCRIPTION
**Pull request by pH5**
Add the required `VkPhysicalDeviceImageDrmFormatModifierInfoEXT` extension structure to the `VkPhysicalDeviceImageFormatInfo2` `pNext` chain when calling `vkGetPhysicalDeviceImageFormatProperties2` with tiling set to `VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT`.

### CODE REVIEW

File: layers/core_checks/cc_device_memory.cpp [hunk 1]
```diff
@@ -212,7 +212,9 @@ bool CoreChecks::HasExternalMemoryImportSupport(const vvl::Buffer &buffer, VkExt
 }
 
 bool CoreChecks::HasExternalMemoryImportSupport(const vvl::Image &image, VkExternalMemoryHandleTypeFlagBits handle_type) const {
-    VkPhysicalDeviceExternalImageFormatInfo external_info = vku::InitStructHelper();
+    VkPhysicalDeviceImageDrmFormatModifierInfoEXT drm_format_modifier_info = vku::InitStructHelper();
```
Comment (artem-lunarg) [inline layers/core_checks/cc_device_memory.cpp:215]: a) In similar places we have a condition that tiling is DRM_FORMAT_MODIFIER. Should the same be done here? https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/d26b50b03815ff226e6df478b4ddc4b98d8deaee/layers/core_checks/cc_image.cpp#L206

b) In similar place we also initialized other fields of `VkPhysicalDeviceImageDrmFormatModifierInfoEXT`:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/d26b50b03815ff226e6df478b4ddc4b98d8deaee/layers/core_checks/cc_image.cpp#L245

Comment (spencer-lunarg) [reply layers/core_checks/cc_device_memory.cpp:215]: I think this is causing the `NegativeExternalMemorySync.ImportMemoryHandleType` test to fail

**Review by artem-lunarg — APPROVED** [review]
LGTM

**Comment by spencer-lunarg** [comment]
@pH5 thanks for the help with the DRM fixes, they are always appreciated!

**Pull request merged**